mappings/doab.py

```diff
@@ -1,130 +1,177 @@
-        self.record.rights = self.parseRights()
+    def _get_authors(self, record, namespaces):
+        datacite_authors = record.xpath('./datacite:creator/text()', namespaces=namespaces)
+        dc_authors = record.xpath('./dc:creator/text()', namespaces=namespaces)
 
-        # Clean up links
-        self.record.has_part = self.parseLinks()
+        authors = datacite_authors or dc_authors
 
-        if self.record.source_id is None or len(self.record.identifiers) < 1:
-            self.raiseMappingError('Malformed DOAB record')
+        return [f'{author}|||true' for author in authors if author is not None or author == '']
 
-    def parseIdentifiers(self):
-        outIDs = []
+    def _get_identifers(self, record, namespaces):
+        dc_ids = record.xpath('./dc:identifier/text()', namespaces=namespaces)
+        datacite_ids = record.xpath('./datacite:identifier/text()', namespaces=namespaces)
+        
+        datacite_alt_ids = self._get_text_type_data(record, namespaces, './datacite:alternateIdentifier', '{}|{}')
+        dc_alt_ids = self._get_text_type_data(record, namespaces, './dc:alternateIdentifier', '{}|{}')
 
-        for iden in self.record.identifiers:
+        ids = [f'{id}|doab' for id in dc_ids + datacite_ids]
+
+        if datacite_alt_ids:
+            ids.extend(datacite_alt_ids)
+
+        if dc_alt_ids:
+            ids.extend(dc_alt_ids)
+
+        new_ids = []
+        source_id = None
+
+        for id in ids:
             try:
-                value, auth = iden.split('|')
+                value, auth = id.split('|')
             except ValueError:
                 continue
 
-            if value[:4] == 'http':
-                doabDOIGroup = re.search(self.DOI_REGEX, value)
-
-                if doabDOIGroup:
-                    value = doabDOIGroup.group(1)
-                    self.record.source_id = value
+            if value.startswith('http'):
+                doab_doi_group = re.search(self.DOI_REGEX, value)
+                if doab_doi_group:
+                    value = doab_doi_group.group(1)
+                    source_id = value
                 else:
                     continue
 
-            outIDs.append('{}|{}'.format(value, auth.lower()))
-
-        return outIDs
+            new_ids.append(f'{value}|{auth.lower()}')
 
-    def parseRights(self):
-        for rightsObj in self.record.rights:
-            rightsData = [d.strip() for d in list(rightsObj.split('|'))]
+        return new_ids, source_id
 
-            if rightsData[1] == '': continue
+    def _get_contributors(self, record, namespaces):
+        return self._get_text_type_data(record, namespaces, './datacite:contributor', '{}|||{}')
 
-            return '|'.join(rightsData)
+    def _get_dates(self, record, namespaces):
+        datacite_dates = self._get_text_type_data(record, namespaces, './datacite:date', '{}|||{}')
+        dc_dates = self._get_text_type_data(record, namespaces, './dc:date', '{}|||{}')
```

Comment:
Here I recommend that if we do not know the date type, we may want to exclude it so 

```
{2021-04-20T15:29:32Z|||,2021-04-20T15:29:32Z|||,2012|||Issued}
```
becomes
```
{2012|||Issued}
```

Reply:
Updated _get_text_type_data to do this since this also makes sense for the other fields we are calling it for (alternative identifiers and contributors)